sdk/containerregistry/container-registry/src/containerRegistryChallengeHandler.ts

```diff
@@ -45,7 +47,12 @@ export class ChallengeHandler implements ChallengeCallbacks {
     });
   }
 
-  authorizeRequest(_options: AuthorizeRequestOptions): Promise<void> {
+  authorizeRequest(options: AuthorizeRequestOptions): Promise<void> {
+    // Try using the existing token in case we don't need to refresh.
+    if (this.cachedAcrAccessToken) {
+      options.request.headers.set("Authorization", `Bearer ${this.cachedAcrAccessToken}`);
```

Comment:
Yeah that's right. If the cached one is invalid (or doesn't have the correct scope), we get a 401 back from the service anyway and handle it with `authorizeRequestOnChallenge` as we would do if we weren't sending a token at all. Using the cached token saves us __a lot__ of requests when uploading in chunks, since previously we were waiting for the challenge response for every single request, which effectively doubles the number of requests being made.